javascript/ql/src/Security/CWE-078/IndirectCommandInjection.qhelp

```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE qhelp PUBLIC
+"-//Semmle//qhelp//EN"
+"qhelp.dtd">
+<qhelp>
+	<overview>
+
+		<p>
+
+			Forwarding command-line arguments to
+			<code>require('child_process').exec</code>, or some other library
+			routine that executes a system command can change the meaning of the
+			command unexpectedly due to unescaped special characters.
+
+		</p>
+
+		<p>
+
+			When the forwarded command-line arguments come from a parent
+			process that has not escaped the special characters in the arguments,
+			then the parent process may indirectly be vulnerable to command-line
+			injection since the special characters gets evaluated unexpectedly.
+
+		</p>
+
+	</overview>
+	<recommendation>
+
+		<p>
+
+			If possible, use hard-coded string literals to specify the
+			command to run or library to load. Instead of forwarding the
+			command-line arguments to the process, examine the command-line
+			arguments and then choose among hard-coded string literals.
+
+		</p>
+
+		<p>
+
+			If the applicable libraries or commands cannot be determined
+			at compile time, then add code to verify that the each forwarded
+			command-line argument is properly escaped before using it.
+
+		</p>
+
+		<p>
+
+			If the forwarded command-line arguments are part of the
+			arguments of the system command, prefer a library routine that handles
+			the arguments as an array of strings rather than a single concatenated
+			string. This prevents the unexpected evaluation of special characters.
+
+		</p>
+
+	</recommendation>
+	<example>
+
+		<p>
+
+			The following example shows code that executes another
```

Comment:
I think it would be clearer if we gave this piece of code a name, or at least have a way of referring to it that distinguishes it (a) from the other JavaScript file it executes, and (b) its (potential) parent process. Something simple like "wrapper script" might do.